src/audio/module_adapter/iadk/iadk_module_adapter.cpp

```diff
@@ -57,6 +57,9 @@ uint32_t IadkModuleAdapter::IadkModuleAdapter_Process(struct sof_source **source
 			i_size = source_get_data_available(sources[i]);
 			ret = source_get_data(sources[i], i_size, (const void **)&input,
 					      (const void **)&input_start, &input_end);
+			if (ret != 0)
```

Comment:
there is type wrong for ret, it should not be uint32_t, it should be int. 
since ret maybe negative value, check source_get_data code below:
https://github.com/thesofproject/sof/blob/main/src/module/audio/source_api.c#L18